### PR TITLE
[genesis] Remove init() in genesis.go

### PR DESCRIPTION
**Pull request by Liuhaai**
# Description
1. Rename `genesis.Default` to `genesis.DefaultConfig`
2. Replace `genesis.Default` in the tessts with `genesis.TestConfig()`

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: blockchain/filedao/testing.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func testVerifyChainDB(t *testing.T, fd FileDAO, start, end uint64) {
 }
 
 func createTestingBlock(builder *block.TestingBuilder, height uint64, h hash.Hash256) *block.Block {
-	block.LoadGenesisHash(&genesis.Default)
+	block.LoadGenesisHash(&genesis.DefaultConfig)
```
Comment (dustinxie) [inline blockchain/filedao/testing.go:227]: can also use `TestConfig()` since this func is for test?

**Review by dustinxie — APPROVED** [review]

File: blockchain/genesis/genesis.go [hunk 2]
```diff
@@ -113,13 +105,18 @@ func defaultConfig() Genesis {
...
 }
 
-// TestDefault is the default genesis config for testing
-func TestDefault() Genesis {
+// TestConfig is the genesis config for testing
+func TestConfig() Genesis {
 	ge := defaultConfig()
 	initTestDefaultConfig(&ge)
 	return ge
 }
```
Comment (CoderZhi) [inline blockchain/genesis/genesis.go:113]: for test only purpose code, it is better to be placed in testutil if possible.

File: blockchain/genesis/genesis.go [hunk 3]
```diff
@@ -113,13 +105,18 @@ func defaultConfig() Genesis {
 	}
 }
 
-// TestDefault is the default genesis config for testing
-func TestDefault() Genesis {
+// TestConfig is the genesis config for testing
+func TestConfig() Genesis {
```
Comment (CoderZhi) [inline blockchain/genesis/genesis.go:109]: The function name doesn't explicitly tell that `initTestDefaultConfig` will be called.

File: blockchain/genesis/heightupgrade_test.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 func TestNewHeightChange(t *testing.T) {
 	require := require.New(t)
 
-	cfg := Default
+	cfg := TestConfig()
```
Comment (CoderZhi) [inline blockchain/genesis/heightupgrade_test.go:17]: revert

File: blockchain/integrity/integrity_test.go [hunk 5]
```diff
@@ -1078,10 +1082,12 @@ func TestConstantinople(t *testing.T) {
 	cfg.Chain.ChainDBPath = testDBPath
 	cfg.Chain.IndexDBPath = testIndexPath
 	cfg.Chain.ProducerPrivKey = "a000000000000000000000000000000000000000000000000000000000000000"
+	cfg.Genesis = genesis.TestConfig()
 	cfg.Genesis.EnableGravityChainVoting = false
 	cfg.Plugins[config.GatewayPlugin] = true
 	cfg.Chain.EnableAsyncIndexWrite = false
 	cfg.ActPool.MinGasPriceStr = "0"
+	cfg.Genesis = genesis.TestConfig()
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:1090]: why do this again, same as L1085 above?

File: blockchain/integrity/integrity_test.go [hunk 6]
```diff
@@ -1335,10 +1341,12 @@ func TestLoadBlockchainfromDB(t *testing.T) {
 	cfg.Chain.TrieDBPath = testTriePath
 	cfg.Chain.ChainDBPath = testDBPath
 	cfg.Chain.IndexDBPath = testIndexPath
+	cfg.Genesis = genesis.TestConfig()
 	cfg.Genesis.EnableGravityChainVoting = false
 	cfg.ActPool.MinGasPriceStr = "0"
 	genesis.SetGenesisTimestamp(cfg.Genesis.Timestamp)
-	block.LoadGenesisHash(&genesis.Default)
+	ge := genesis.TestConfig()
+	block.LoadGenesisHash(&ge)
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:1349]: would `block.LoadGenesisHash(&cfg.Genesis)` work?

File: blockchain/integrity/integrity_test.go [hunk 7]
```diff
@@ -1440,9 +1449,9 @@ func TestBlockchainInitialCandidate(t *testing.T) {
 		blockchain.BlockValidatorOption(sf),
 	)
 	rolldposProtocol := rolldpos.NewProtocol(
-		genesis.Default.NumCandidateDelegates,
-		genesis.Default.NumDelegates,
-		genesis.Default.NumSubEpochs,
+		genesis.TestConfig().NumCandidateDelegates,
+		genesis.TestConfig().NumDelegates,
+		genesis.TestConfig().NumSubEpochs,
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:1454]: `cfg.Genesis.Numxxx`

File: consensus/scheme/rolldpos/rolldposctx_test.go [hunk 8]
```diff
@@ -48,9 +48,9 @@ func TestRollDPoSCtx(t *testing.T) {
 	})
 
 	rp := rolldpos.NewProtocol(
-		genesis.Default.NumCandidateDelegates,
-		genesis.Default.NumDelegates,
-		genesis.Default.NumSubEpochs,
+		genesis.TestConfig().NumCandidateDelegates,
+		genesis.TestConfig().NumDelegates,
+		genesis.TestConfig().NumSubEpochs,
```
Comment (dustinxie) [inline consensus/scheme/rolldpos/rolldposctx_test.go:53]: g.Numxxx

**Comment by raullenchai** [comment]
Close b/c it has been inactive for 1+yr.

**Pull request closed without merging**